s3transfer/futures.py

```diff
@@ -128,8 +128,9 @@ def set_result(self, result):
         Implies that the TransferFuture succeeded.
         """
         with self._lock:
-            self._result = result
-            self._status = 'success'
+            if self._exception is None:
+                self._result = result
+                self._status = 'success'
```

Comment:
Not intentional. More of an oversight of what happens after a task is completed.